README.md

```diff
@@ -1,55 +1,105 @@
+  * [Running a particular unit/integration test with log4j output](#running-a-particular-unitintegration-test-with-log4j-output)
+  * [Specifying test retries](#specifying-test-retries)
+* [Running a Kafka broker in KRaft mode](#running-a-kafka-broker-in-kraft-mode)
+  * [Using compiled files](#using-compiled-files)
+  * [Using docker image](#using-docker-image)
+* [Test Coverage](#test-coverage)
+  * [Generating test coverage reports](#generating-test-coverage-reports)
+* [Other Actions](#other-actions)
+  * [Building a binary release gzipped tarball](#building-a-binary-release-gzipped-tar-ball)
+  * [Building auto generated messages](#building-auto-generated-messages)
+  * [Cleaning the build](#cleaning-the-build)
+  * [Running a task for a specific project](#running-a-task-for-a-specific-project)
+  * [Listing all gradle tasks](#listing-all-gradle-tasks)
+  * [Building IDE project](#building-ide-project)
+  * [Publishing the streams quickstart archetype artifact to maven](#publishing-the-streams-quickstart-archetype-artifact-to-maven)
+  * [Installing specific projects to the local Maven repository](#installing-specific-projects-to-the-local-maven-repository)
+  * [Building the test jar](#building-the-test-jar)
+* [Running code quality checks](#running-code-quality-checks)
+  * [Checkstyle](#checkstyle)
+  * [Spotless](#spotless)
+  * [Spotbugs](#spotbugs)
+* [JMH microbenchmarks](#jmh-microbenchmarks)
+* [Dependency Analysis](#dependency-analysis)
+  * [Determining if any dependencies could be updated](#determining-if-any-dependencies-could-be-updated)
+* [Common build options](#common-build-options)
+* [Running system tests](#running-system-tests)
+* [Running in Vagrant](#running-in-vagrant)
+* [Contributing Kafka](#contributing-kafka)
+  
+# Building Kafka 
+
+[**Apache Kafka**](https://kafka.apache.org) is an open-source distributed event streaming platform used by thousands of companies for high-performance data pipelines, streaming analytics, data integration, and mission-critical applications.
+
+## Prerequisites <a name="prerequisites"></a>
+* You need to have [Java 8+](http://www.oracle.com/technetwork/java/javase/downloads/index.html) installed.
+* KRaft or ZooKeeper
+* Apache Kafka is built and tested using Java versions 17 and 23. However, to ensure maximum compatibility, the Kafka client and streams modules are compiled to work with Java 11 and later, while the broker and tools require Java 17 or later.
+* Scala 2.13 is the only supported version in Apache Kafka.
+
+## Build steps <a name="build-steps"></a>
+### Build a jar and run it <a name="build-a-jar-and-run-it"></a>
     ./gradlew jar
+* Follow instructions in [**Kafka Quickstart**](https://kafka.apache.org/quickstart)
 
-Follow instructions in https://kafka.apache.org/quickstart
-
-### Build source jar ###
+### Build source jar <a name="build-source-jar"></a>
```

Comment:
These empty links seem superfluous, markdown already has a system for linking sections together.